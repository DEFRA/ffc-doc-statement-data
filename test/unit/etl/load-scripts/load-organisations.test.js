const { loadOrganisations } = require('../../../../app/etl/load-scripts/load-organisations')

jest.mock('../../../../app/data', () => ({
  sequelize: {
    query: jest.fn()
  }
}))

jest.mock('../../../../app/etl/load-scripts/load-interm-utils')
jest.mock('../../../../app/config/message', () => ({
  day0DateTime: null
}))

describe('loadOrganisations', () => {
  let mockTransaction

  beforeEach(() => {
    mockTransaction = {}
  })

  test('should call executeQuery with correct parameters when day0DateTime is not set', async () => {
    const startDate = new Date('2023-01-01')

    const { executeQuery } = require('../../../../app/etl/load-scripts/load-interm-utils')
    executeQuery.mockResolvedValueOnce()

    await loadOrganisations(startDate, mockTransaction)

    expect(executeQuery).toHaveBeenCalledWith(expect.stringContaining('INSERT INTO public.organisations'), {
      startDate
    }, mockTransaction)
  })
})
