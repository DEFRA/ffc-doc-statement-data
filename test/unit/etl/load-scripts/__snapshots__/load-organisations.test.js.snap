// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadOrganisations should call sequelize.query with correct parameters 1`] = `
[MockFunction] {
  "calls": [
    [
      "
    INSERT INTO public.organisations (
      sbi, "addressLine1", "addressLine2",
      "addressLine3", city, county,
      postcode, "emailAddress", frn,
      "name", updated
    )
    SELECT DISTINCT ON (sbi)
      sbi, "addressLine1", "addressLine2",
      "addressLine3", city, county,
      SUBSTRING(postcode, 1, 7), "emailAddress", frn::integer,
      "name", NOW()
    FROM public."etlIntermOrg" O
    WHERE O."etlInsertedDt" > :startDate
    ORDER BY sbi, "etlInsertedDt" DESC;
  ",
      {
        "raw": true,
        "replacements": {
          "startDate": 2023-01-01T00:00:00.000Z,
        },
        "transaction": {},
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
